backend/typescript/services/implementations/volunteerService.ts

```diff
@@ -29,26 +33,63 @@ class VolunteerService implements IVolunteerService {
     return {
       id: newVolunteer.id,
       userId: String(newVolunteer.user_id),
+      status: newVolunteer.status,
     };
   }
 
-  async getVolunteerByID(volunteerId: string): Promise<UserVolunteerDTO> {
+  async getVolunteerById(volunteerId: string): Promise<UserVolunteerDTO> {
+    let volunteer: Volunteer | null;
+    let user: User | null;
+
+    try {
+      volunteer = await Volunteer.findByPk(Number(volunteerId), {
+        include: User,
+      });
+
+      if (!volunteer) {
+        throw new Error(`volunteerId ${volunteerId} not found.`);
+      }
+
+      user = volunteer.user;
+      if (!user) {
+        throw new Error(`userId ${volunteer.user_id} not found.`);
+      }
+    } catch (error: unknown) {
```

Comment:
nit: don't think we need this
```suggestion
    } catch (error) {
```